qc_baselib/models/result.py

```diff
@@ -23,15 +23,35 @@ class XMLLocationType(BaseXmlModel):
     xpath: str = attr(name="xpath")
 
 
+class MessageLocationType(BaseXmlModel):
+    index: int = attr(name="index")
+    channel: Optional[str] = attr(name="channel", default=None)
+    field: Optional[str] = attr(name="field", default=None)
+    time: Optional[float] = attr(name="time", default=None)
+
+
+class TimeLocationType(BaseXmlModel):
+    time: float = attr(name="time")
+
+
 class InertialLocationType(BaseXmlModel):
     x: float = attr(name="x")
     y: float = attr(name="y")
     z: float = attr(name="z")
 
 
 class FileLocationType(BaseXmlModel):
-    column: int = attr(name="column")
-    row: int = attr(name="row")
+    column: Optional[int] = attr(name="column", default=None)
+    row: Optional[int] = attr(name="row", default=None)
+    offset: Optional[int] = attr(name="offset", default=None)
+
+    @model_validator(mode="after")
+    def check_at_least_one_attribute(self) -> Any:
+        if self.column is None and self.row is None and self.offset is None:
+            raise ValueError(
+                "FileLocationType requires at least one of the attributes: column, row, or offset"
+            )
+        return self
```

Comment:
```suggestion
        if any((getattr(self, field) for field in self.model_fields_set())):
           return self
        raise ValueError(
           f"FileLocationType requires at least one of the attributes: {self.modelf_fields_set()}")
```

Should be robust with respect to additional fields, if FileLocationType changes again. But not tested! @pmai please check it before accepting suggestion (or trash it altogether).